### PR TITLE
Import github.com/canonical/go-efilib

### PR DESCRIPTION
**Pull request by chrisccoulson**
This makes use of github.com/canonical/go-efilib, which tcglog-parser also pulls in, and replaces some of the functionality that we had in this repository.

### CODE REVIEW

File: efi/secureboot_policy_test.go [hunk 1]
```diff
@@ -340,43 +130,49 @@ func TestComputeDbUpdate(t *testing.T) {
...
+			newSignatures: []int{77},
 		},
 		{
 			desc:          "AppendMsDbxUpdate/2",
 			orig:          "testdata/efivars2/dbx-d719b2cb-3d3a-4596-a3bc-dad00e67656f",
 			update:        "testdata/updates1/dbx/MS-2016-08-08.bin",
 			quirkMode:     SigDbUpdateQuirkModeDedupIgnoresOwner,
 			sha1hash:      testutil.DecodeHexStringT(t, "96f7dc104ee34a0ce8425aac20f29e2b2aba9d7e"),
-			newSignatures: 77,
+			newESLs:       1,
```
Comment (pedronis) [inline efi/secureboot_policy_test.go:142]: does it make sense to have tests where this is > 1 ?

**Review by pedronis — APPROVED** [review]
+1,  one question

Skimming: also the for loop using isNewSig could maybe also use a "break Label",  it seems it needs the bool anyway, but it could be slightly more readable

File: efi/secureboot_policy_test.go (hunk 1 above)
Comment (chrisccoulson) [reply efi/secureboot_policy_test.go:142]: I've added the 2020 dbx update now, which has more than 1 ESL.

**Comment by chrisccoulson** [comment]
> +1, one question
> 
> Skimming: also the for loop using isNewSig could maybe also use a "break Label", it seems it needs the bool anyway, but it could be slightly more readable

I've added a label to the outer loop now.

File: internal/efi/efi.go [hunk 2]
```diff
@@ -19,7 +19,13 @@
...
 
+import (
+	"github.com/canonical/go-efilib"
+)
+
 var (
 	EventLogPath = "/sys/kernel/security/tpm0/binary_bios_measurements" // Path of the TCG event log for the default TPM, in binary form
 	EFIVarsPath  = "/sys/firmware/efi/efivars"                          // Default mount point for efivarfs
+
+	ReadVar = efi.ReadVar
```
Comment (bboozzoo) [inline internal/efi/efi.go:30]: For a minute there I though that we could update snapd to use go-efilib too, but we only read the efi variables so it's probably not really worth it.

**Review by bboozzoo — APPROVED** [review]
LGTM

File: efi/secureboot_policy_test.go [hunk 3]
```diff
@@ -160,6 +160,15 @@ func TestComputeDbUpdate(t *testing.T) {
 			newESLs:       1,
 			newSignatures: []int{1},
 		},
+		{
+			desc:          "AppendDbxUpdateWithMultipleESLs",
+			orig:          "testdata/efivars2/dbx-d719b2cb-3d3a-4596-a3bc-dad00e67656f",
+			update:        "testdata/updates5/dbx/dbxupdate_x64_1.bin",
+			quirkMode:     SigDbUpdateQuirkModeNone,
+			sha1hash:      testutil.DecodeHexStringT(t, "306985f67cdc580e25c1572568f9e65e420984d2"),
+			newESLs:       4,
```
Comment (pedronis) [inline efi/secureboot_policy_test.go:169]: thanks

**Review by pedronis — APPROVED** [review]
still +1

**Pull request merged**